src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2750,40 +2821,29 @@ private void AddEnumMembers(MembersAndInitializersBuilder result, EnumDeclaratio
             }
         }
 
-        private static void AddInitializer(ref ArrayBuilder<FieldOrPropertyInitializer>? initializers, ref int aggregateSyntaxLength, FieldSymbol? fieldOpt, CSharpSyntaxNode node)
+        private static void AddInitializer(ref ArrayBuilder<FieldOrPropertyInitializer.Builder>? initializers, FieldSymbol? fieldOpt, CSharpSyntaxNode node)
```

Comment:
>ArrayBuilder [](start = 47, length = 12)

Consider `ImmutableArray<FieldOrPropertyInitializer.Builder>.Builder?` to make it clear that we're not using pooled instances, so we won't have to worry about freeing. That would also propagate to `AddNonTypeMembers`, `AddInitializers` and `MembersAndInitializersBuilder.InstanceInitializersForRecordDeclarationWithParameters`

Or, if we can remove `InstanceInitializersForRecordDeclarationWithParameters`, then can we use pooled instances (freeing in `AddInitializers`)?

nit: consider making this a local function in `AddNonTypeMembers`.

Reply:
I don't think it is worth making changes that you suggest. Besides none of that is added in this PR. 

---
In reply to: [434399118](https://github.com/dotnet/roslyn/pull/44803#discussion_r434399118) [](ancestors = 434399118)